app/src/main/java/com/jerboa/ui/components/home/HomeActivity.kt

```diff
@@ -344,6 +353,18 @@ fun MainPostListingsContent(
             showPostLinkPreviews = showPostLinkPreviews,
             openImageViewer = appState::toView,
             openLink = appState::openLink,
+            markAsReadOnScroll = markAsReadOnScroll,
+            onMarkAsRead = { postView ->
+                account?.also { acct ->
+                    homeViewModel.markPostAsRead(
+                        MarkPostAsRead(
+                            post_id = postView.post.id,
+                            read = true,
+                            auth = acct.jwt,
+                        ),
+                    )
+                }
+            },
```

Comment:
All these need to check first if the post is already read ifso dont do the markAsRead, this saves some api calls